packages/storage/src/client_storage/mod.rs

```diff
@@ -27,6 +27,10 @@ macro_rules! set_dir {
         #[cfg(not(target_family = "wasm"))]
         $crate::set_directory(std::path::PathBuf::from($path))
     };
+    ($path:expr) => {
```

Comment:
Rather than having an expr to be used freely on `set_directory`, why not just extend the current `$path:literal` to use an expr?

Something like

```
    ($path: expr) => {
        #[cfg(not(target_family = "wasm"))]
        $crate::storage::set_directory(std::path::PathBuf::from(format!("{}", $path)))
    };
```

Reply:
`std::path::PathBuf::from($path)` should work with `Path`s, `PathBuf`s and `&str`